client/src/components/Account/Friends/FriendsTab.tsx

```diff
@@ -1,9 +1,113 @@
-type Props = {}
+import { useState, useEffect } from 'react';
+import {
+  useGetUserFriendQuery,
+  useDeleteFriendMutation,
+} from '../../../redux/services/user/userService';
+import { User as UserType } from '../../../types';
+import { useNavigate } from 'react-router-dom';
+import { Button, Modal } from 'flowbite-react';
+import { HiOutlineExclamationCircle } from 'react-icons/hi';
+function FriendsTab() {
+  const { data, isLoading } = useGetUserFriendQuery('User');
+  const [friends, setFriends] = useState<UserType[]>();
+  const [deleteFriend] = useDeleteFriendMutation();
+  const [openModal, setOpenModal] = useState<string | undefined>();
+  const [tempFriend, setTempFriend] = useState<string>('');
 
-function FriendsTab({}: Props) {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (data && !isLoading) {
+      setFriends(data);
+    }
+  }, [data, isLoading]);
+  // let tempFriend : string | undefined = undefined;
+  const onFriendClicked = (id: string) => {
+    navigate(`/auth/compareSchedule/${id}`);
+  };
+  const onConfirmFriendDeleted = async () => {
+    try {
+      await deleteFriend({ friendID: tempFriend });
+      setOpenModal(undefined);
+    } catch (error: any) {
+      console.log(error);
+    }
+  };
+  const tempFriendHandler = (id: string) => {
+    setTempFriend(id);
+    setOpenModal('pop-up');
+  };
   return (
-    <div>FriendsTab</div>
-  )
+    <div className="flex flex-col items-center justify-center gap-6 rounded dark:text-white">
+      <>
+        <Modal
+          show={openModal === 'pop-up'}
```

Comment:
nitpicky: Same thing here with the magic strings.

Reply:
This part, actually isn't allowing me with flowbite-UI